File: svnrepo/RIT_JBF_source/src/ECSearchButton.java
/*
 * Copyright (C) 2013 Rochester Institute of Technology
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU General Public License as
 * published by the Free Software Foundation; either version 2 of the
 * License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
 *
 * See the GNU General Public License for more details.
 */
import java.awt.Canvas;
import java.awt.Font;
import java.awt.Image;
import java.awt.Rectangle;
import java.awt.Graphics;
import java.awt.Color;
import java.awt.Cursor;
import java.awt.event.MouseListener;
import java.awt.event.MouseEvent;
/**
 * Searches for EC number of proteins whose function is enzyme
 *
 * @author Jill Zapoticznyj
 *
 * Much of this code is based off of the RemoveProteinButton class
 * @author Adam Bazinet
 *
 */
public class ECSearchButton extends Canvas implements MouseListener {

    private Electro2D electro2D;
    private Color bgColor = Color.BLACK;
    private Color fillColorOff = Color.RED;
    private Color fillColorOn = new Color(255,165,0); //orange
    private boolean highlighted = false;
    private Color textColor = Color.white;
    private Font textFont = new Font("Arial", Font.BOLD, 14);
    private Rectangle d = null;  //dimensions of this component
    private String pro_id = "";  //the ID of the protein

    //double buffering variables
    private Image buffer = null;
    private Graphics bufferGraphics = null;

    /**
     * Constructor, performs some perfunctory tasks.
     *
     * @param e - a reference to Electro2D
     * @param id - the string being searched for
     */
    public ECSearchButton( Electro2D e, String id ) {
	electro2D = e; //give the button a reference to Electro2D
	pro_id = id; //set the search value to the string passed to the method
	//have the button register itself as a MouseListener in order to
	//respond to mouse events from the user
	this.addMouseListener(this);
    }
    
    /**
     * Paints user controls.
     */
    public void paint(Graphics g) {

	//initialize buffering variables
	if(d == null || buffer == null || bufferGraphics == null) {
	    d = getBounds();
	    //set image size equal to component size
	    buffer = createImage(d.width, d.height);
	    bufferGraphics = buffer.getGraphics();
	}
	
	//draw to the buffer
	bufferGraphics.setColor(bgColor);
	bufferGraphics.fillRect(0, 0, d.width, d.height);
	// change the background color depending on the location of the cursor
	if(highlighted == true) {
	    bufferGraphics.setColor(fillColorOn);
	} else {
	    bufferGraphics.setColor(fillColorOff);
	}
	bufferGraphics.fillRoundRect(0, 0, d.width-1, d.height-1, 15, 15);
	bufferGraphics.setColor(textColor);
	bufferGraphics.setFont(textFont);
	bufferGraphics.drawString("Search for Enzyme Info", 8, 14);

	//finally, draw image onto Graphics object
	g.drawImage(buffer, 0, 0, this);

    }
    public void update(Graphics g) {
	paint(g);
    }

    /**
     * Mouse listener methods. Used to respond to user button presses, etc.
     */
    public void mousePressed(MouseEvent e) {
	//do nothing
    }
    public void mouseReleased(MouseEvent e) {
	//do nothing
    }
    /**
     * Responds to the cursor being placed over this button.
     */
    public void mouseEntered(MouseEvent e) {
	//change the image of the cursor
	setCursor(new Cursor(Cursor.HAND_CURSOR));
	highlighted = true; //register that the cursor is over the button
	//and redraw the button to reflect the change in bg color
	repaint();
	//display a message in the status bar reflecting the purpose of this
	// button.
	//electro2D.showStatus("Perform a SwissProt search for this protein.");
    }
    
    /**
     * Responds to the mouse being removed from over the button.
     */
    public void mouseExited(MouseEvent e) {
	//change the image of the cursor
	setCursor(new Cursor(Cursor.DEFAULT_CURSOR));
	highlighted = false; //register that the cursor is not over the button
	//and redraw the button to reflect the change in bg color
	repaint();
	//remove the message from the status bar
	//electro2D.showStatus("");
    }
    /**
     * Responds to the user clicking on the button.
     */
    public void mouseClicked(MouseEvent e) {
	//open the search page and perform the search for the protein
	electro2D.showECSearchPage( pro_id );
    }

}

File: site/swingjs/j2s/main/java/Electro2D/ECSearchButton.js
(function(){var P$=Clazz.newPackage("main.java.Electro2D"),I$=[[0,'java.awt.Color','java.awt.Font','java.awt.Cursor']],I$0=I$[0],$I$=function(i,n){return((i=(I$[i]||(I$[i]=Clazz.load(I$0[i])))),!n&&i.$load$&&Clazz.load(i,2),i)};
/*c*/var C$=Clazz.newClass(P$, "ECSearchButton", null, 'java.awt.Canvas', 'java.awt.event.MouseListener');

C$.$clinit$=2;

Clazz.newMeth(C$, '$init$', function () {
this.bgColor=$I$(1).BLACK;
this.fillColorOff=$I$(1).RED;
this.fillColorOn=Clazz.new_($I$(1,1).c$$I$I$I,[255, 165, 0]);
this.highlighted=false;
this.textColor=$I$(1).white;
this.textFont=Clazz.new_($I$(2,1).c$$S$I$I,["Arial", 1, 14]);
this.d=null;
this.pro_id="";
this.buffer=null;
this.bufferGraphics=null;
},1);

C$.$fields$=[['Z',['highlighted'],'S',['pro_id'],'O',['electro2D','main.java.Electro2D.Electro2D','bgColor','java.awt.Color','+fillColorOff','+fillColorOn','+textColor','textFont','java.awt.Font','d','java.awt.Rectangle','buffer','java.awt.Image','bufferGraphics','java.awt.Graphics']]]

Clazz.newMeth(C$, 'c$$main_java_Electro2D_Electro2D$S',  function (e, id) {
Clazz.super_(C$, this);
this.electro2D=e;
this.pro_id=id;
this.addMouseListener$java_awt_event_MouseListener(this);
}, 1);

Clazz.newMeth(C$, 'paint$java_awt_Graphics',  function (g) {
if (this.d == null  || this.buffer == null   || this.bufferGraphics == null  ) {
this.d=this.getBounds$();
this.buffer=this.createImage$I$I(this.d.width, this.d.height);
this.bufferGraphics=this.buffer.getGraphics$();
}this.bufferGraphics.setColor$java_awt_Color(this.bgColor);
this.bufferGraphics.fillRect$I$I$I$I(0, 0, this.d.width, this.d.height);
if (this.highlighted == true ) {
this.bufferGraphics.setColor$java_awt_Color(this.fillColorOn);
} else {
this.bufferGraphics.setColor$java_awt_Color(this.fillColorOff);
}this.bufferGraphics.fillRoundRect$I$I$I$I$I$I(0, 0, this.d.width - 1, this.d.height - 1, 15, 15);
this.bufferGraphics.setColor$java_awt_Color(this.textColor);
this.bufferGraphics.setFont$java_awt_Font(this.textFont);
this.bufferGraphics.drawString$S$I$I("Search for Enzyme Info", 8, 14);
g.drawImage$java_awt_Image$I$I$java_awt_image_ImageObserver(this.buffer, 0, 0, this);
});

Clazz.newMeth(C$, 'update$java_awt_Graphics',  function (g) {
this.paint$java_awt_Graphics(g);
});

Clazz.newMeth(C$, 'mousePressed$java_awt_event_MouseEvent',  function (e) {
});

Clazz.newMeth(C$, 'mouseReleased$java_awt_event_MouseEvent',  function (e) {
});

Clazz.newMeth(C$, 'mouseEntered$java_awt_event_MouseEvent',  function (e) {
this.setCursor$java_awt_Cursor(Clazz.new_($I$(3,1).c$$I,[12]));
this.highlighted=true;
this.repaint$();
});

Clazz.newMeth(C$, 'mouseExited$java_awt_event_MouseEvent',  function (e) {
this.setCursor$java_awt_Cursor(Clazz.new_($I$(3,1).c$$I,[0]));
this.highlighted=false;
this.repaint$();
});

Clazz.newMeth(C$, 'mouseClicked$java_awt_event_MouseEvent',  function (e) {
this.electro2D.showECSearchPage$S(this.pro_id);
});

Clazz.newMeth(C$);
})();
;Clazz.setTVer('3.3.1-v1');//Created 2021-10-03 19:41:55 Java2ScriptVisitor version 3.3.1-v1 net.sf.j2s.core.jar version 3.3.1-v1
